tools/BUILD.bazel

```diff
@@ -111,3 +111,70 @@ refresh_compile_commands(
         "//sel/...": "--extra_toolchains=//tools:clang_toolchain",
     },
 )
+
+genrule(
+    name = "gen-coverage",
+    srcs = ["@lcov//:genhtml"],
+    outs = ["coverage.bash"],
+    cmd = """
+echo "#!/bin/bash" > $@
+echo "set -euo pipefail" >> $@
+echo "" >> $@
+echo "cd \\$$BUILD_WORKSPACE_DIRECTORY" >> $@
+echo "COVERAGE_DIR={cov_dir}" >> $@
+echo "bazelisk coverage \\$${{@:-//...}}" >> $@
+echo "$(execpath @lcov//:genhtml) {options} {report}" >> $@
+echo "" >> $@
+echo "echo wrote coverage report to" >> $@
+echo "echo file://{html}" >> $@
+""".format(
+        cov_dir = "\\$$(bazelisk info output_path)/_coverage",
+        html = "\\$${COVERAGE_DIR}/html/index.html",
+        options = " ".join([
+            "--output-directory \\$${COVERAGE_DIR}/html",
+            "--show-details",
+            "--function-coverage",
+            "--branch-coverage",
+            "--title zipline",
+            "--highlight",
+            "--legend",
+            "--missed",
+            "--demangle-cpp",
+            "--dark-mode",
+        ]),
+        report = "\\$${COVERAGE_DIR}/_coverage_report.dat",
+    ),
+)
+
+sh_binary(
+    name = "coverage",
+    srcs = [":coverage.bash"],
+    args = [
+        "--instrumentation_filter=//sel",
+        "//...",
+    ],
+    data = ["@lcov//:genhtml"],
+)
+
+_coverage_open_script = """
+echo "#!/bin/bash" > $@
+echo "set -euo pipefail" >> $@
+echo "" >> $@
+echo "cd \\$$BUILD_WORKSPACE_DIRECTORY" >> $@
+echo "exec {{cmd}} {html}" >> $@
```

Comment:
This needs to invoke the earlier script to ensure that the coverage report is created